docs/source/src/rust/user-guide/expressions/expression-expansion.rs

```diff
@@ -0,0 +1,215 @@
+    // --8<-- [end:all]
+
+    // --8<-- [start:all-exclude]
+    let result = df
+        .clone()
+        .lazy()
+        .select([all().exclude(["^day_.*$"])])
+        .collect()?;
+    println!("{}", result);
+    // --8<-- [end:all-exclude]
+
+    // --8<-- [start:col-exclude]
+    let result = df
+        .clone()
+        .lazy()
+        .select([dtype_col(&DataType::Float64).exclude(["^day_.*$"])])
+        .collect()?;
+    println!("{}", result);
+    // --8<-- [end:col-exclude]
+
+    // --8<-- [start:duplicate-error]
+    let gbp_usd_rate = 1.31; // As of 14th October 2024
+
+    let result = df
+        .clone()
+        .lazy()
+        .select([
+            col("price") / lit(gbp_usd_rate),
+            col("price") / lit(eur_usd_rate),
+        ])
+        .collect();
+    match result {
+        Ok(df) => println!("{}", df),
+        Err(e) => println!("{}", e),
+    };
+    // --8<-- [end:duplicate-error]
+
+    // --8<-- [start:alias]
+    let _result = df
+        .clone()
+        .lazy()
+        .select([
+            (col("price") / lit(gbp_usd_rate)).alias("price (GBP)"),
+            (col("price") / lit(eur_usd_rate)).alias("price (EUR)"),
+        ])
+        .collect()?;
+    // --8<-- [end:alias]
+
+    // --8<-- [start:prefix-suffix]
+    let result = df
+        .clone()
+        .lazy()
+        .select([
+            (col("^year_.*$") / lit(eur_usd_rate))
+                .name()
+                .prefix("in_eur_"),
+            (cols(["day_high", "day_low"]) / lit(gbp_usd_rate))
+                .name()
+                .suffix("_gbp"),
+        ])
+        .collect()?;
+    println!("{}", result);
+    // --8<-- [end:prefix-suffix]
+
+    // --8<-- [start:name-map]
+    // There is also `name().to_uppercase()`, so this usage of `map` is moot.
+    let result = df
+        .clone()
+        .lazy()
+        .select([all()
+            .name()
+            .map(|name| Ok(PlSmallStr::from_string(name.to_ascii_uppercase())))])
+        .collect()?;
+    println!("{}", result);
+    // --8<-- [end:name-map]
+
+    // --8<-- [start:for-with_columns]
+    let mut result = df.clone().lazy();
+    for tp in ["day", "year"] {
+        let high = format!("{}_high", tp);
+        let low = format!("{}_low", tp);
+        let aliased = format!("{}_amplitude", tp);
+        result = result.with_column((col(high) - col(low)).alias(aliased))
+    }
+    let result = result.collect()?;
+    println!("{}", result);
+    // --8<-- [end:for-with_columns]
+
+    // --8<-- [start:yield-expressions]
+    let mut exprs: Vec<Expr> = vec![];
+    for tp in ["day", "year"] {
+        let high = format!("{}_high", tp);
+        let low = format!("{}_low", tp);
+        let aliased = format!("{}_amplitude", tp);
+        exprs.push((col(high) - col(low)).alias(aliased))
+    }
+    let result = df.clone().lazy().with_columns(exprs).collect()?;
+    println!("{}", result);
+    // --8<-- [end:yield-expressions]
```

Comment:
If someone could please check that this code looks reasonable (within the context of the guide “Expressions > Expression expansion > Programmatically generating expressions”) I would be glad, since I have very little experience with Rust.

Reply:
Looks good to me :+1: